src/psyclone/psyir/frontend/fortran.py

```diff
@@ -60,12 +64,37 @@ def psyir_from_source(self, source_code):
 
         :returns: PSyIR representing the provided Fortran source code.
         :rtype: :py:class:`psyclone.psyir.nodes.Node`
+
         '''
         string_reader = FortranStringReader(source_code)
         parse_tree = self._parser(string_reader)
         psyir = self._processor.generate_psyir(parse_tree)
         return psyir
 
+    def psyir_from_literal_expression(self, source_code):
+        '''
+        Generate the PSyIR tree for the supplied Fortran literal expression.
+
+        :param str source_code: text of the expression to be parsed.
+
+        :returns: PSyIR representing the provided Fortran expression.
+        :rtype: :py:class:`psyclone.psyir.nodes.Node`
+
+        :raises NotImplementedError: if the supplied expression contains \
+                                     any variable symbols.
+        '''
+        parse_tree = Fortran2003.Expr(source_code)
```

Comment:
Are there source_code inputs that could make this statement fail? Is it worth a try-except around it?

Can you add in the unit tests some examples with inputs that don't necessarily have symbols but they can not be found in an expression (e.g. a return statement, and if statement comparing literals, ...). Also a test with a non-expression and a symbol, to test what error happens first (e.g. passing a full assignment). And then a test with just gibberish which is not Fortran at all.

Reply:
Oops, that was a bit of an omission. Fixed it now. There is a bit of a problem in that e.g. "return" is rejected as containing a symbol rather than because it is not an expression. However, I argue we can postpone tweaking that until we add support for symbols (should we ever need to).